File: Mixter.Domain.Tests/Identity/UserIdTest.cs
using Mixter.Domain.Identity;
using NFluent;
using Xunit;

namespace Mixter.Domain.Tests.Identity
{
    public class UserIdTest
    {
        private const string UserEmail = "test@mixit.fr";

        [Fact]
        public void WhenCreate2IdWithSameEmailsThenAreEquals()
        {
            var userId1 = new UserId(UserEmail);
            var userId2 = new UserId(UserEmail);

            Check.That(userId1).IsEqualTo(userId2);
        }

        [Fact]
        public void WhenToStringIdThenEmail()
        {
            var userId1 = new UserId(UserEmail);

            Check.That(userId1.ToString()).IsEqualTo(UserEmail);
        }

        [Fact]
        public void WhenCreateUserIdWithEmptyEmailThenThrowUserEmailCannotBeEmpty()
        {
            Check.ThatCode(() => new UserId(string.Empty)).Throws<UserEmailCannotBeEmpty>();
        }
    }
}

File: domain/src/test/java/mixter/domain/identity/UserIdTest.java
package mixter.domain.identity;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class UserIdTest {

    public static final String AN_EMAIL = "mail@mix-it.fr";

    @Test
    public void GivenAnEmailWhenCreatingAUserIdThenAUserIdShouldBeCreate() throws Exception {
        UserId userId1 = new UserId(AN_EMAIL);
    }
    @Test
    public void Given2UserIdCreatedFromTheSameEmailWhenComparedWithEqualsThenTheyShouldBeEqual() throws Exception {
        UserId userId1 = new UserId(AN_EMAIL);
        UserId userId2 = new UserId(AN_EMAIL);

        assertThat(userId1).isEqualTo(userId2);
    }

    @Test
    public void GivenAnExistingUserIdWhenSerializingItToStringThenItShouldSerializeAsTheEmailString() throws Exception {
        //Given
        UserId userId1 = new UserId(AN_EMAIL);
        //When
        String seralizedUserId = userId1.toString();
        //Then
        assertThat(seralizedUserId).isEqualTo(AN_EMAIL);
    }
}
